engine/src/main/battlecode/world/InternalRobot.java

```diff
@@ -360,25 +372,26 @@ public void empower(int radiusSquared) {
         if (numBots == 0)
             return;
         
-        int convictionToGive = (int) (this.conviction * this.gameWorld.getTeamInfo().getBuff(this.team));
+        long convictionToGive = (long) (this.conviction * this.gameWorld.getTeamInfo().getBuff(this.team));
```

Comment:
doesn't this compute product as an int, then cast that int to long?
think ((long) this.conviction) * ((long) this.gameWorld...) is better

Reply:
I could be wrong, maybe casting goes the other way

Reply:
I think `getBuff` returns a double so this computation upcasts to `double`, and then downcasts to `long`.

Reply:
can cast first term to long to be safe

Reply:
https://repl.it/@n8kim1/EvenHarmfulBoards#Main.java